EventFilter/EcalTBRawToDigi/src/EcalTB07DaqFormatter.cc

```diff
@@ -217,7 +217,7 @@ void EcalTB07DaqFormatter::interpretRawData(const FEDRawData & fedData ,
     
     
     short TowerStatus[MAX_TT_SIZE+1];
-    char buffer[20];
+    char buffer[25];
```

Comment:
this is needed apparently for ```sprintf(buffer, "FE_CHSTATUS#%d", i);```
and the error message is ```EcalTB07DaqFormatter.cc:224:10: note: 'sprintf' output between 14 and 23 bytes into a destination of size 20```
There are 12 characters in the format, "i" can be up to 2 (it's in the range of 1 to 70), and then there is a termination "0" for a total of 15 chars.
What's going on?
Is this added "to be safe just because an arbitrary int can take up to MAX_INT"?
e.g. (impossible in this context) "FE_CHSTATUS#-2147483648" will take 23 chars +1 for termination.
This looks like a false-positive to me and we'd likely have to make these follow-up PRs every time someone uses a similar pattern

Reply:
Yes. Compiler calculates the highest possible buffer size based on the types & format specifiers. We cleaned such things in the past, but new flags revealed a few more places we missed. In particularly this would be very bad if `sprintf` is used. E.g. due to some mistake (a bug) `i` overflows and then `sprintf` damages data on the buffer because it overflows buffer.

Technically in this case we can just use `std::to_string`:
```
auto blah = "FE_CHSTATUS#" + std::to_string(10);
```

Note ,this might be interesting also: https://github.com/cms-sw/cmssw/issues/21281